tools/database/src/memtrie.rs

```diff
@@ -78,3 +93,158 @@ impl LoadMemTrieCommand {
         Ok(())
     }
 }
+
+#[derive(clap::Parser)]
+pub struct SplitShardTrieCommand {
+    #[clap(long)]
+    shard_uid: ShardUId,
+    #[clap(long)]
+    boundary_account: AccountId,
+}
+
+impl SplitShardTrieCommand {
+    pub fn run(
+        &self,
+        home: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        let env_filter = EnvFilterBuilder::from_env().verbose(Some("memtrie")).finish()?;
+        let _subscriber = default_subscriber(env_filter, &Default::default()).global();
+        let mut near_config = nearcore::config::load_config(&home, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+        near_config.config.store.load_memtries_for_tracked_shards = true;
+        let genesis_config = &near_config.genesis.config;
+
+        let rocksdb = Arc::new(open_rocksdb(home, near_store::Mode::ReadOnly)?);
+        let store = near_store::NodeStorage::new(rocksdb).get_hot_store();
+
+        let epoch_manager =
+            EpochManager::new_arc_handle(store.clone(), &genesis_config, Some(home));
+        let runtime =
+            NightshadeRuntime::from_config(home, store.clone(), &near_config, epoch_manager)
+                .context("could not create the transaction runtime")?;
+        let final_head = runtime.store().chain_store().final_head()?;
+
+        let shard_layout = ShardLayout::derive_shard_layout(
+            &genesis_config.shard_layout,
```

Comment:
you should get the current shard layout from the epoch manager rather than using the genesis one

Reply:
Fixed.